pulsar-flink-connector/src/main/java/org/apache/flink/streaming/connectors/pulsar/formats/protobufnative/PulsarProtobufNativeFormatFactory.java

```diff
@@ -105,17 +105,34 @@ public ChangelogMode getChangelogMode() {
     }
 
     private String extractTopicName(DynamicTableFactory.Context context) {
-
         validateTopic(context.getCatalogTable().getOptions());
-        String topic = context.getCatalogTable().getOptions().get(TOPIC.key());
-        // Maybe catalog mode
-        if (StringUtils.isNullOrWhitespaceOnly(topic)) {
-            final ObjectIdentifier table = context.getObjectIdentifier();
-            topic = TopicName.get(table.getDatabaseName() + "/" + table.getObjectName()).toString();
+
+        String topic;
+        if (isGenericTable(context)) {
```

Comment:
There is a isGenericDatabase() method in PulsarCatalogSupport, which uses a different checking method

Reply:
That's a catalog internal implementation. 

Other code should not assume any of its implementation. Just use the `isGeneric` table options, which is exposed as an indicator.